redo cobalt purity system

planetme opened the pull request:
can someone test this pls

planetme committed:
redo cobalt purity system

trainvoi commented on groovy/material/FirstDegreeMaterials.groovy:
```diff
@@ -1789,7 +1787,7 @@ public class FirstDegreeMaterials {
                 .components(Cassiterite)
                 .colorAverage()
-                .iconSet(DULL)
+                .iconSet(DULL)f
```
typo

trainvoi commented on groovy/postInit/chemistry/inorganic_chemistry/Magnets.groovy:
```diff
@@ -69,7 +69,7 @@ mods.gregtech.electrolyzer.recipeBuilder()
+        .inputs(ore('dustHighPurityCobalt') * 5)
         .fluidInputs(fluid('argon') * 300)
         .fluidOutputs(fluid('samarium_alloy') * 864)
         .blastFurnaceTemp(1800)
```
should we make SmCo ABS recipe auto-generated by blastTemp property in SmCo material registration?

trainvoi replied:
alright, the manually-registered ABS recipe needs to be removed now that ABS recipes are auto-generated now.

trainvoi replied:
@planetme this recipe needs to be removed.

trainvoi commented on groovy/postInit/metallurgy/Superalloys.groovy:
```diff
@@ -5,14 +5,14 @@ ABS = recipemap('alloy_blast_smelter')
-    .inputs(ore('dustNiobium') * 1)
-    .inputs(ore('dustTantalum') * 1)
-    .inputs(ore('dustMolybdenum') * 2)
+    .inputs(ore('dustAnyPurityIron') * 12)
```
these recipes are removed by #1376. We might need a mixin to make the ABS uses anyPurity dusts.

trainvoi commented on groovy/material/FirstDegreeMaterials.groovy:
```diff
@@ -1770,14 +1770,12 @@ public class FirstDegreeMaterials {
 
 
-        ImpureCobaltOxide = new Material.Builder(8360, SuSyUtility.susyId('impure_cobalt_oxide'))
+        CobaltOxide = new Material.Builder(8360, SuSyUtility.susyId('cobalt_oxide'))
```
isn't CobaltOxide already registered by GTCEu?

planetme committed:
fixes

planetme committed:
make smco abs recipe autogen

trainvoi committed:
fixes

- removing redundant material: ImpureCobaltSulfateSolution
- move CobaltSulfateSolution and register HighPurityCobaltSulfateSolution to SecondDegreeMaterials.groovy
- register HighPurityCobaltSulfateSolution to SuSyMaterials.groovy
- register Alloy Smelter recipe map in Catalysts.groovy

trainvoi committed:
Update SecondDegreeMaterials.groovy

planetme committed:
remove smco recipe

planetme committed:
Merge branch 'master-ceu' into cobalt-rework

trainvoi commented:
1.) how can players get HP cobalt oxide
2.) how are players suppose to electrolyze their first cobalt when they haven't got their first Co electrode

bruberu reviewed (dismissed):
There definitely needs to be a pure cobalt oxide recipe, given how the hydrocracking catalyst now needs it.
![image](https://github.com/user-attachments/assets/96cfc33a-5b72-4490-8ef0-bc6986bb6c37)

planetme committed:
fix everything

trainvoi reviewed (dismissed):
lgtm

trainvoi reviewed (approved):
approve again because i didn't notice that bru's comment got resolved already
